include/control_filters/exponential_filter.hpp

```diff
@@ -0,0 +1,133 @@
+
+#include <memory>
+#include <limits>
+#include <string>
+
+#include "filters/filter_base.hpp"
+
+#include "control_toolbox/filters.hpp"
+#include "exponential_filter_parameters.hpp"
+
+namespace control_filters
+{
+
+/***************************************************/
+/*! \class ExponentialFilter
+  \brief A exponential filter class.
+
+  This class implements a low-pass filter for
+  various data types.
+
+  \section Usage
+
+  The ExponentialFilter class is meant to be instantiated as a filter in
+  a controller but can also be used elsewhere.
+  For manual instantiation, you should first call configure()
+  (in non-realtime) and then call update() at every update step.
+
+*/
+/***************************************************/
+
+template <typename T>
+class ExponentialFilter : public filters::FilterBase<T>
+{
+public:
+  /*!
+   * \brief Configure the ExponentialFilter (access and process params).
+   */
+  bool configure() override;
+
+  /*!
+   * \brief Applies one iteration of the exponential filter.
+   *
+   * \param data_in input to the filter
+   * \param data_out filtered output
+   *
+   * \returns false if filter is not configured, true otherwise
+   */
+  bool update(const T & data_in, T & data_out) override;
+
+private:
+  rclcpp::Clock::SharedPtr clock_;
+  std::shared_ptr<rclcpp::Logger> logger_;
+  std::shared_ptr<exponential_filter::ParamListener> parameter_handler_;
+  exponential_filter::Params parameters_;
+  T last_smoothed_value;
+};
+
+template <typename T>
+bool ExponentialFilter<T>::configure()
+{
+  logger_.reset(
+    new rclcpp::Logger(this->logging_interface_->get_logger().get_child(this->filter_name_)));
+
+  // Initialize the parameters once
+  if (!parameter_handler_)
+  {
+    try
+    {
+      parameter_handler_ =
+        std::make_shared<exponential_filter::ParamListener>(this->params_interface_,
+                                                         this->param_prefix_);
+    }
+    catch (rclcpp::exceptions::ParameterUninitializedException & ex) {
+      RCLCPP_ERROR((*logger_), "Exponential filter cannot be configured: %s", ex.what());
+      parameter_handler_.reset();
+      return false;
+    }
+    catch (rclcpp::exceptions::InvalidParameterValueException & ex)  {
+      RCLCPP_ERROR((*logger_), "Exponential filter cannot be configured: %s", ex.what());
+      parameter_handler_.reset();
+      return false;
+    }
+  }
+  parameters_ = parameter_handler_->get_params();
+
+  last_smoothed_value = std::numeric_limits<double>::quiet_NaN();
+
+  return true;
+}
+
+template <typename T>
+bool ExponentialFilter<T>::update(const T & data_in, T & data_out)
+{
+  if (!this->configured_)
+  {
+    throw std::runtime_error("Filter is not configured");
+  }
+
+  // Update internal parameters if required
+  if (parameter_handler_->is_old(parameters_))
+  {
+    parameters_ = parameter_handler_->get_params();
+  }
+
+  if (std::isnan(last_smoothed_value))
```

Comment:
```suggestion
  if (!std::isfinite(last_smoothed_value))
```

Reply:
isn' this legitimate here as we initialize this variable with `std::numeric_limits<T>::quiet_NaN()`?

Reply:
Sure. it is fine